src/main/java/zd/zero/waifu/motivator/plugin/service/WaifuGatekeeper.kt

```diff
@@ -0,0 +1,44 @@
+package zd.zero.waifu.motivator.plugin.service
+
+import com.intellij.openapi.Disposable
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.components.ServiceManager
+import zd.zero.waifu.motivator.plugin.settings.PluginSettingsListener
+import zd.zero.waifu.motivator.plugin.settings.WaifuMotivatorPluginState
+import zd.zero.waifu.motivator.plugin.settings.WaifuMotivatorState
+
+class WaifuGatekeeper : Disposable {
+    companion object {
+        val instance: WaifuGatekeeper
+            get() = ServiceManager.getService(WaifuGatekeeper::class.java)
+    }
+
+    private val connection = ApplicationManager.getApplication().messageBus.connect()
+
+    private var allowedWaifu: Set<String> = constructList(WaifuMotivatorPluginState.getPluginState())
+
+    private fun constructList(pluginState: WaifuMotivatorState): Set<String> =
+        pluginState.preferredCharacters.split(WaifuMotivatorState.DEFAULT_DELIMITER)
+            .filter { it.isNotEmpty() }
+            .map { it.toLowerCase() }
+            .toSet()
+
+    init {
+        connection.subscribe(PluginSettingsListener.PLUGIN_SETTINGS_TOPIC, object : PluginSettingsListener {
+            override fun settingsUpdated(newPluginState: WaifuMotivatorState) {
+                allowedWaifu = constructList(newPluginState)
+            }
+        })
+    }
+
+    fun isAllowed(characters: List<String>?): Boolean =
+        allowedWaifu.isEmpty() ||
+            characters?.any { allowedWaifu.contains(it.toLowerCase()) } ?: false
```

Comment:
Does this method intended to be used?

Reply:
https://github.com/zd-zero/waifu-motivator-plugin/blob/a1d19fb7a2610de062bdc951d7bf2b9952c4684a/src/main/java/zd/zero/waifu/motivator/plugin/assets/VisualAssetManager.kt#L18

Or did I not understand correctly? Was this a question about how the method should work?

Reply:
Oh my, my IDE is an impostor...

It's flagging as never used :/
<img width="720" alt="Screen Shot 2020-10-16 at 07 39 51" src="https://user-images.githubusercontent.com/21978370/96196748-cb8b4800-0f82-11eb-98f5-ae1449df66ba.png">